src/ThemedStyleSheet.js

```diff
@@ -17,10 +25,12 @@ function create(makeFromTheme, createWithDirection) {
 }
 
 function createLTR(makeFromTheme) {
+  themeChangedLTR = false;
```

Comment:
Per our offline conversation, this global toggle will create an issue if multiple components have been rendered before the new theme is registered. The first component will recreate its styles, flipping the global toggle, and the rest of the components will fail to recreate their styles. So, it likely makes sense to bring this logic back into `withStyles`, so it's handled with a component-level flag.